WalletWasabi/Extensions/MemoryExtensions.cs

```diff
@@ -13,7 +13,7 @@ public static class MemoryExtensions
 
 		private static object AsyncLocksLock { get; } = new object();
 
-		public static async Task<TItem> AtomicGetOrCreateAsync<TItem>(this IMemoryCache cache, object key, Func<ICacheEntry, Task<TItem>> factory)
+		public static async Task<TItem> AtomicGetOrCreateAsync<TItem>(this IMemoryCache cache, object key, MemoryCacheEntryOptions options, Func<Task<TItem>> factory)
```

Comment:
I would suggest to use an factory method for the options to avoid creating the options again and again in every call.
```suggestion
		public static async Task<TItem> AtomicGetOrCreateAsync<TItem>(this IMemoryCache cache, object key, Func<MemoryCacheEntryOptions> optionsFactory, Func<Task<TItem>> factory)
```

Reply:
NACK, makes it harder to use.

Reply:
I would not use an extension class in the first place. A simple class that accepts a cache instance and options sound better to me.